### ISSUE TITLE
Remove ToxDNS and related stuff from toxcore

### ISSUE THREAD

**Issue by ovalseven8** [irungentoo/toxcore#1491]
Hello,

as far as I know, there's the consensus that toxcore should remove ToxDNS at all.

In general, the goal of toxcore is to provide a lighweight, reliable and secure codebase for the clients. The current solution with ToxDNS isn't both secure and decentralized.
Moreover, to use Tox IDs is not such a problem as it probably seems in my opinion. Nonetheless, the clients can of course implement HTTPS lookup services - but that's something that shouldn't be in toxcore.

Are there any plans when to remove ToxDNS and all the related stuff from toxcore?

**Comment by GrayHatter** [comment]
@ovalseven8 once it can be replaced with something better.

**Comment by ovalseven8** [comment]
I'm not sure whether that justifies to keep it in toxcore.
As already said, the clients can implement it as well.

**Comment by subliun** [comment]
@GrayHatter you don't think HTTPS lookups are better than DNS?

**Comment by cebe** [comment]
whats the benefit of https over dns? (assuming dnssec is supported for security)

**Comment by GrayHatter** [comment]
Https is inherently more secure, because it's been built to be secure.
Rather than security being an after thought (dnssec).

@subliun no, I think https is also broken. But you know that so stop
trolling github

**Comment by subliun** [comment]
@cebe https is encrypted, dnssec isn't. 

@GrayHatter In what way to you feel https is broken?

**Comment by GrayHatter** [comment]
Https was better because it'd be easier for anyone to create/run a resolve
server. Because it'd be backed by the trust chain already installed on a
users computer. The problem is that you'd have to trust everyone who can
issue a trusted cert. That's fine for web traffic, but Tox needs a higher
level of security. The solution implemented by toxme is just as bad as as
DNS where you have to have the key for each service you want to trust with
the added deficit of https giving a false sense of security.

**Comment by LuccoJ** [comment]
@GrayHatter why can't you only trust specific certificates? Just because web browsers have a comprehensive list of certificate authorities doesn't have to mean Tox has to use the same list, it could use a much more narrow one.

No matter what type of encryption and digital signature you're going to you, you will have to trust *some* signing authority, whether it's built into the client or external.

**Comment by GrayHatter** [comment]
@LuccoJ because defining your own cert list is much harder than using the existing on built into the OS, and as I said HTTPS was supposed to be easy. But if you're doing that, why not DNS?

Right, but a huge part of tox is to be trustless (just ask @JFreegman ). I'm fine with trust, but I'm not fine with trusting every TLA. So if we want to be the IM client that protects its users, why HTTPS?

**Comment by LuccoJ** [comment]
> Right, but a huge part of tox is to be trustless (just ask @JFreegman ). I'm fine with trust, but I'm not fine with trusting every TLA. So if we want to be the IM client that protects its users, why HTTPS?

This will be a problem with *any* centralized (or even federated) naming system: trustlessness will be impossible. The only real alternative (and I do wish this was pursued) is to use a distributed system, such as for instance the most obvious choice of basing it on namecoin.
However, although I realize blockchain-based systems are not "lightweight", there is no shortcut: it's either that, or a system where you trust someone who's certifying something.
Nobody has found another way to solve Zooko's triangle yet.

**Comment by GrayHatter** [comment]
I'm fine with trust. 

I'm not fine with trusting [your TLA here]

I'm fine with distributed, centralized, decentralized and federated. 

I'm not fine with a blockchain.

**Comment by LuccoJ** [comment]
You seem to be fine with a few things that the Tox project at least originally was not. Personally I hope it stays the original way.

**Comment by GrayHatter** [comment]
@LuccoJ have you read every line of code in toxcore? And then every line in the client that you use?

Then you're already using trust.

You use github, and I assume IRC, so you've also fine with both decentralized and centralized systems as well. 

What exactly do you think I'm fine with that's in opposition to the original idea of Toxcore?

**Comment by LuccoJ** [comment]
I'm using distributed trust when I use toxcore, because even though I haven't read every line of code in the client I use, it is possible for everyone to scrutinize it.

I use GitHub and IRC but that doesn't mean I'm "fine" with them. I find GitHub's centralization very worrying, and the worry has been materializing lately because they have been reorganizing their management in ways that could signal trouble.
I find the fact that open source, sometimes distributed projects use a centralized, partly closed service like GitHub to host their code quite ironic (but perhaps unavoidable for now: there are possible future alternatives, like IPFS or Sia; look them up).

IRC is pretty much terrible both in terms of reliability and security, whether or not you use it in a centralized way (freenode-style) or in a federated way (like most other networks).

The reason I'm interested in Tox is because is different from this sort of legacy, not because it's the same.

I'm also interested in many other distributed systems, of which there is a long list at https://github.com/redecentralize/alternative-internet
Tox isn't unique (see Ring), and will only remain interesting if it stays true to its promise of distributedness, not if it becomes a dubious mixed system.

**Comment by GrayHatter** [comment]
Right, but that was my point. I'm fine with them as long as they are the right tool for the job.

And github can do anything it likes, because if they do anything shady, FOSS will jump ship. Once that happens github will stop existing. Meanwhile, they can't do anything directly with code because git is very resistant to that. 

Just like Toxcore is and should be. If you're using toxme.io, or register.utox.org, you're already using centralized systems. With the requirement that you trust them.

Trust, and centralized systems aren't bad. Just easy to abuse.

**Comment by subliun** [comment]
@GrayHatter so can dns be removed from core now?

**Comment by JFreegman** [comment]
Yes

**Comment by subliun** [comment]
@Jfreegman \o/
